src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -348,7 +358,14 @@ ThreadError DatasetManager::Set(Coap::Header &aHeader, Message &aMessage, const
 
         aMessage.Read(offset, sizeof(Tlv), &data.tlv);
         aMessage.Read(offset + sizeof(Tlv), data.tlv.GetLength(), data.value);
-        mLocal.Set(data.tlv);
+
+        if (data.tlv.GetType() != Tlv::kChannel && data.tlv.GetType() != Tlv::kPanId &&
+            data.tlv.GetType() != Tlv::kNetworkMasterKey &&
+            data.tlv.GetType() != Tlv::kMeshLocalPrefix)
+        {
+            mLocal.Set(data.tlv);
```

Comment:
I don't think the Leader should not be breaking up the Active Dataset like this.  Datasets as received from other Thread devices need to be apply all or nothing.  If the Active Dataset contains TLVs that affect connectivity, the entire Active Dataset should be delayed via the Pending Dataset and the current Active Dataset should only be delayed after the Delay Timer expires.  Please let me know if you think the spec states otherwise.